core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2650,6 +2678,28 @@ private[zio] trait ZIOFunctions extends Serializable {
   final def untraced[R, E, A](zio: ZIO[R, E, A]): ZIO[R, E, A] =
     zio.untraced
 
+  /**
+   * Feeds elements of type A to f and accumulates all errors in error channel or successes in success channel.
+   *
+   * This combinator is lossy meaning that if there are errors all successes will be lost.
+   * To retain all information please use ZIO#partitionM
+   */
+  final def validateM[R, E, A, B](in: Iterable[A])(f: A => ZIO[R, E, B]): ZIO[R, List[E], List[B]] =
```

Comment:
We should require `implicit ev: CanFail[E]` for each of these methods since they don't make sense for effects that cannot fail.